_posts/2018-01-24-kubernetes-and-hokusai.markdown

```diff
@@ -0,0 +1,134 @@
+---
+layout: post
+title: "In the 'Whelp!' of the Great Wave"
+date: 2018-01-24
+comments: true
+author: ash
+categories: [kubernetes, hokusai, open-source, danger]
+---
+
+This past week has found me working on a brand new Rails project. Now, if I was building this project for my personal needs, I would without a doubt deploy it to [Heroku][] – for both the ease of use and the high level of abstraction that [Dynos][] afford. But I'm not building this for myself, I'm building it for my team.
+
+<!-- more -->
+
+While Heroku is easy to get started with, costs scale up quickly. And, as described in our [2017 tech stack post][stack], our team is moving more and more towards [Kubernetes][]. I had almost no experience with Kubernetes before last week, and I was intimidated by the Kubernetes web UI. With some help from my colleague Isac, who wrote the [Hokusai][] tool, I was able to get a staging environment up and running in under a day.
+
+But let's step back first.
+
+My background is in iOS software development, so spinning up new servers isn't something I do often. When I _do_, I usually use Heroku. After deploying to it, it feels like Kubernetes is a kind of hosted Heroku: it handles scaling up instances, managing worker/db/other instances, load-balancers, environment variables, promoting from staging to production – all that stuff. But Kubernetes' sophistication comes with a sophisticated user interface. 
+
+So basically, Kubernetes is to Heroku what Hokusai is to the Heroku command-line tool.
```

Comment:
Hokusai was very much conceived with this intention!  I am very happy the point came across, without me actually stating it ^_^  However, maybe the wording order... "Hokusai is to Kubernetes what the Heroku CLI is to the Heroku platform"?

Reply:
Agreed – I struggled with the wording there, I like that better.